Add support for grafted DIDs whose chaintrees don't yet exist

cap10morgan opened the pull request:
Are there other `GetTip` errors we should worry about? Can we even tell the difference?

cap10morgan committed:
Add support for grafted DIDs whose chaintrees don't yet exist

brandonwestcott commented on graftabledag/graftabledag.go:
```diff
@@ -47,7 +49,7 @@ func New(origin *dag.Dag, dagGetter DagGetter) (*GraftedDag, error) {
 func (gd *GraftedDag) getChaintreeDag(ctx context.Context, did string) (*dag.Dag, error) {
 	tip, err := gd.dagGetter.GetTip(ctx, did)
 	if err != nil {
```
I do think we should be explicit about the errors. What i was running into in dgit was `hamt.ErrNotFound`, which will be returned for [server](https://github.com/quorumcontrol/tupelo/blob/78e6495685970f57c5cd1aea5fea5e90f79063be/sdk/gossip/types/graftedownership.go#L108) and [client](https://github.com/quorumcontrol/tupelo/blob/78e6495685970f57c5cd1aea5fea5e90f79063be/sdk/gossip/types/graftedownership.go#L52).

Maybe this can be left up to the `dagGetter`s though, maybe they just return a known error (like `TipNotFound`) and then let it propagate here?

cap10morgan replied:
Good suggestions. Making it so.

cap10morgan replied:
That's pushed here and to the corresponding tupelo PR too: https://github.com/quorumcontrol/tupelo/pull/474

cap10morgan committed:
Define top-level ErrTipNotFound

This will be used in Tupelo to indicate specific error conditions we want to handle differently (e.g. grafted ownership before the owning chaintree exists).
